sklearn/titanic/README.md

```diff
@@ -0,0 +1,99 @@
+## Introduction
+
+This sample code illustrates how to start a GCE instance and train a model using scikit-learn on the instance. We will be using the [Titanic dataset](https://www.kaggle.com/c/titanic) from [Kaggle](https://www.kaggle.com/). The goal is to train a classifier to predict whether a given passenger on Titanic survived or not.
+
+## Prerequisites
+
+You need to download [train.csv](https://www.kaggle.com/c/titanic/download/train.csv) from [Kaggle](https://www.kaggle.com/).
+
+The sample assumes that you already have a [GCP](https://cloud.google.com/) account and you have [downloaded](https://cloud.google.com/sdk/), installed, and [configured](https://cloud.google.com/sdk/gcloud/reference/config/) it on your computer.
+
+Finally, you need to have a project created on GCP. For further instructions, please check [here](https://cloud.google.com/sdk/gcloud/reference/projects/create).
+
+
+## Steps:
+We will create a Google Compute Engine instance, configure it, and use it to train a model. While some of the steps can be done using the web portal [here](https://pantheon.corp.google.com), we will try to accomplish this using the SDK and mainly the [gcloud](https://cloud.google.com/sdk/gcloud/) command.
+
+##### 1. Create a Google Compute Engine Instance
+First, we should create a GCE instance that we can use to train our model on. In this case, we can use many of the default arguments and simply create the instance:
+
+```
+gcloud compute instances create MYINSTANCE
+```
+
+where MYINSTANCE is our given name to this instance. By default, this instance will be of type *n1-standard-1* with 1 CPU and 3.75GB of RAM. To see a list of available machine types, you can use:
+
+```
+gcloud compute machine-types list
+```
+
+By default, the instance is going to be built using a Debian image. But you can change this during the instance creation. To see a list of available images, you may use the following command:
+```
+gcloud compute images list
+```
+
+##### 2. Accessing and Configuring the Instance
+
+Accessing the newly created image is quite simple:
+```
+gcloud compute ssh MYINSTANCE
+```
+
+Once we ssh to the instance, we will need to use *pip* to install scikit-learn and some other packages:
+```
+# Run on the new instance
+sudo apt-get install python-pip
+pip install numpy pandas sklearn scipy tensorflow
+```
+We are not using tensorflow to train the model in this sample. However we will be using *gfile*, a python class implemented in tensorflow which unifies how we access the local and remote files.
+
+If you want to use GCS to store the dataset, the new instance will then require to obtain user access credentials. The following command will guide you through the process:
+```
+# Run on the new instance
+gcloud auth application-default login
```

Comment:
@happyhuman : This will use your user credentials on the instance, right?

I think we are better off giving the instance service account the right permissions rather than having it treat itself as the user. Does this introduce a lot of additional friction, do you think?